sp-files/contributing.rst

```diff
@@ -0,0 +1,233 @@
+
+
+Licence and copyright
+---------------------
+
+By default, all contributions to ACME are made under the AGPLv3 licence.
+See the `licence <https://github.com/canonical/acme/blob/main/COPYING>`_
+in the ACME GitHub repository for details.
+
+All contributors must sign the `Canonical contributor licence agreement
+<https://ubuntu.com/legal/contributors>`_,
+which grants Canonical permission to use the contributions.
+The author of a change remains the copyright owner of their code
+(no copyright assignment occurs).
+
+
+Pull requests
+-------------
+
+Submit your changes as pull requests on `GitHub
+<https://github.com/canonical/acme>`_.
+Before you open a pull request,
+ensure that the build occurs without any warnings or errors;
+this includes documentation and tests.
+
+Your changes will be reviewed in due time;
+if approved, they will be eventually merged.
+
+
+PR description
+~~~~~~~~~~~~~~
+
+.. TODO: Update with your template checklist details or drop if excessive
+
+- **Title**: Summarise the change in a short, descriptive title.
+
+- **Description**: Provide an explanation of the problem your PR addresses.
+  Highlight any new features, bug fixes or refactoring.
+
+- **Relevant issues**: Reference any
+  `related issues, PRs and repos <https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/autolinked-references-and-urls>`_.
+
+- **Testing**: Explain whether new or updated tests are included.
+
+- **Reversibility**: If you propose decisions that may be costly to reverse,
+  outline the reasons and provide the steps to reverse if necessary.
+
+
+Commits
+~~~~~~~
+
+Use separate commits for each logical change,
+and for changes to different components.
+Prefix your commits with the component they affect,
+using the code tree structure,
+e.g. prefix a commit that updates the ACME service with ``acme/service:``.
+
+Use `conventional commits <https://www.conventionalcommits.org/>`_
+for your commit messages to ensure consistency across the project:
+
+.. code-block:: none
+
+   Ensure correct permissions and ownership for the content mounts
+    
+    * Work around an ACME issue regarding empty dirs:
+      https://github.com/canonical/acme/issues/12345
+    
+    * Ensure the source directory is owned by the user running a container.
+
+   Links:
+   - ...
+   - ...
+
+
+Such structure makes it easier to review contributions
+and simplifies porting fixes to other branches.
+
+
+Developer certificate of origin
+-------------------------------
+
+.. TODO: Update with your details or drop if excessive
+
+To improve contribution tracking,
+we use the `DCO 1.1 <https://developercertificate.org/>`_
+and require a "sign-off" for any changes going into each branch.
+
+The sign-off is a simple line at the end of the commit message
+certifying that you wrote it
+or have the right to commit it as an open-source contribution.
+
+To sign off on a commit, use the ``--signoff`` option in ``git commit``.
+
+
+Code
```

Comment:
Because this file is under the `docs/` folder, so I would rather consider it as a Documentation Contributing Guide. 
If we decide to provide guidelines on coding, perhaps we can prompt a message asking the developer/author to review and update the guide as needed

Reply:
The same comment applies to "Code structure" and "Test"